File: GFG/Deque/firstpump.py
def firstpump(arr,brr):
    n=len(arr)
    start=0
    curr=0
    prev=0
    for i in range(n):
        curr+=(arr[i]-brr[i])
        if(curr<0):
            start=i+1
            prev=curr
            curr=0
    return (start+1) if ((curr+prev)>=0) else -1

q=[50,10,60,100]
w=[30,20,100,10]
print(firstpump(q,w))

File: GFG/Deque/firstpump.cpp
#include<iostream>
#include<deque>
#include<vector>
using namespace std;

int firstind(vector<int> &arr,vector<int> &brr){
    int n=arr.size();
    int start=0,curr_pet=0,prev_pet=0;
    for(int i=0;i<n;i++){
        curr_pet+=(arr[i]-brr[i]);
        cout<<curr_pet<<" ";
        if(curr_pet<0){
            start=i+1;
            prev_pet=curr_pet;
            curr_pet=0;
        }
    }

    return ((curr_pet+prev_pet)>=0) ? (start+1) : -1;
}

int main(){
    vector<int> q {4,8,7,4};
    vector<int> w {6,5,3,5};
    cout<<firstind(q,w)<<endl;
    return 0;
}